Exclude FP input devices for rosemary

zainarbani opened the pull request:
rosemary uses 96 for fp custom keycode which detected by InputManager as KEY_KPENTER.
 screenshots key combo will fail if more than 2 key events detected, exclude it.

Signed-off-by: zainarbani <zaintsyariev@gmail.com>

phhusson commented:
Please do it the same way it's done for other Xiaomi devices

28 août 2022 14:41:13 Zain Arbani ***@***.***>:

> rosemary uses 96 for fp custom keycode which detected by InputManager as KEY_KPENTER.
> screenshots key combo will fail if more than 2 key events detected, exclude it.
> 
> Signed-off-by: zainarbani ***@***.***
> 
> ----------------------------------------
> 
> *You can view, comment on, or merge this pull request online at:*
> 
>   https://github.com/phhusson/device_phh_treble/pull/325
> 
> 
> *Commit Summary*
> 
* > ba81cfa[https://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6] Exclude FP input devices for rosemary
> 
> *File Changes*
> 
> (2 files[https://github.com/phhusson/device_phh_treble/pull/325/files])
> 
* > *A* files/rosemary-excluded-input-devices.xml[https://github.com/phhusson/device_phh_treble/pull/325/files#diff-083586316d8479c0b0d5c5996cb9ab961e60f5dad20d277f3b5c6d02a17b4dda] (31)
* > *M* rw-system.sh[https://github.com/phhusson/device_phh_treble/pull/325/files#diff-29b2fa125a8bb33ed2b8f9ce4c2cc149006642b81294b5da700abbc058e21c95] (3)
> 
> *Patch Links:*
> 
* > https://github.com/phhusson/device_phh_treble/pull/325.patch
* > https://github.com/phhusson/device_phh_treble/pull/325.diff
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/phhusson/device_phh_treble/pull/325], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4ORZ7NGQKNKKWRKJTWLV3NM6RANCNFSM573AINGA].
> You are receiving this because you are subscribed to this thread.[Image de pistage][https://github.com/notifications/beacon/AAAA4OWDLCHTKR3L5KUEPMTV3NM6RA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4UFKLURA.gif]
>

zainarbani commented:
Separated ifelse sections?

zainarbani commented:
> Please do it the same way it's done for other Xiaomi devices 28 août 2022 14:41:13 Zain Arbani ***@***.***>:
> rosemary uses 96 for fp custom keycode which detected by InputManager as KEY_KPENTER. screenshots key combo will fail if more than 2 key events detected, exclude it. Signed-off-by: zainarbani ***@***.*** ---------------------------------------- *You can view, comment on, or merge this pull request online at:*   #325 *Commit Summary*
> * > [ba81cfa](https://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6)[[ba81cfa](https://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6)] Exclude FP input devices for rosemary
> *File Changes* (2 files[https://github.com/phhusson/device_phh_treble/pull/325/files])
> * > *A* files/rosemary-excluded-input-devices.xml[https://github.com/phhusson/device_phh_treble/pull/325/files#diff-083586316d8479c0b0d5c5996cb9ab961e60f5dad20d277f3b5c6d02a17b4dda] (31) * > *M* rw-system.sh[https://github.com/phhusson/device_phh_treble/pull/325/files#diff-29b2fa125a8bb33ed2b8f9ce4c2cc149006642b81294b5da700abbc058e21c95] (3)
> *Patch Links:*
> * > https://github.com/phhusson/device_phh_treble/pull/325.patch * > https://github.com/phhusson/device_phh_treble/pull/325.diff
> […](#)
> — Reply to this email directly, view it on GitHub[#325], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4ORZ7NGQKNKKWRKJTWLV3NM6RANCNFSM573AINGA]. You are receiving this because you are subscribed to this thread.[Image de pistage][https://github.com/notifications/beacon/AAAA4OWDLCHTKR3L5KUEPMTV3NM6RA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4UFKLURA.gif]


Separated ifelse sections?

phhusson commented:
No, mount binding key layout

28 août 2022 15:30:36 Zain Arbani ***@***.***>:

> Please do it the same way it's done for other Xiaomi devices 28 août 2022 14:41:13 Zain Arbani /*@*/./*>:
> rosemary uses 96 for fp custom keycode which detected by InputManager as KEY_KPENTER. screenshots key combo will fail if more than 2 key events detected, exclude it. Signed-off-by: zainarbani /*@*/.*/ ---------------------------------------- /You can view, comment on, or merge this pull request online at:/   #325[https://github.com/phhusson/device_phh_treble/pull/325] /Commit Summary/
> 
* > ba81cfa[https://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6][ba81cfa[https://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6]] Exclude FP input devices for rosemary
> /File Changes/ (2 files[https://github.com//pull/325[https://github.com/phhusson/device_phh_treble/pull/325]/files])
> 
* > /A/ files/rosemary-excluded-input-devices.xml[https://github.com//pull/325[https://github.com/phhusson/device_phh_treble/pull/325]/files#diff-083586316d8479c0b0d5c5996cb9ab961e60f5dad20d277f3b5c6d02a17b4dda] (31) * > /M/ rw-system.sh[https://github.com//pull/325[https://github.com/phhusson/device_phh_treble/pull/325]/files#diff-29b2fa125a8bb33ed2b8f9ce4c2cc149006642b81294b5da700abbc058e21c95] (3)
> /Patch Links:/
> 
* > https://github.com/phhusson/device_phh_treble/pull/325.patch * > https://github.com/phhusson/device_phh_treble/pull/325.diff
> …[#]
> — Reply to this email directly, view it on GitHub[#325[https://github.com/phhusson/device_phh_treble/pull/325]], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4ORZ7NGQKNKKWRKJTWLV3NM6RANCNFSM573AINGA]. You are receiving this because you are subscribed to this thread.[Image de pistage][https://github.com/notifications/beacon/AAAA4OWDLCHTKR3L5KUEPMTV3NM6RA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4UFKLURA.gif]
> 
> Separated ifelse sections?
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/phhusson/device_phh_treble/pull/325#issuecomment-1229457612], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4OR3CJX4GXJMUHAKMCDV3NSXXANCNFSM573AINGA].
> You are receiving this because you commented.[Image de pistage][https://github.com/notifications/beacon/AAAA4OXSHNO3FQ5ZFIHMDBTV3NSXXA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOJFEARTA.gif]
>

zainarbani commented:
> No, mount binding key layout 28 août 2022 15:30:36 Zain Arbani ***@***.***>:
> Please do it the same way it's done for other Xiaomi devices 28 août 2022 14:41:13 Zain Arbani /*@*/./*>: rosemary uses 96 for fp custom keycode which detected by InputManager as KEY_KPENTER. screenshots key combo will fail if more than 2 key events detected, exclude it. Signed-off-by: zainarbani /*@*/.*/ ---------------------------------------- /You can view, comment on, or merge this pull request online at:/   #325[#325] /Commit Summary/
> * > [ba81cfa](https://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6)[[ba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bba81cfa%5Bhttps://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6]](https://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bba81cfa%5Bhttps://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6%5D)] Exclude FP input devices for rosemary
> /File Changes/ (2 files[#325]/files])
> * > /A/ files/rosemary-excluded-input-devices.xml[#325]/files#diff-083586316d8479c0b0d5c5996cb9ab961e60f5dad20d277f3b5c6d02a17b4dda] (31) * > /M/ rw-system.sh[#325]/files#diff-29b2fa125a8bb33ed2b8f9ce4c2cc149006642b81294b5da700abbc058e21c95] (3)
> /Patch Links:/
> * > https://github.com/phhusson/device_phh_treble/pull/325.patch * > https://github.com/phhusson/device_phh_treble/pull/325.diff
> […](#)
> …[#] — Reply to this email directly, view it on GitHub[#325[#325]], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4ORZ7NGQKNKKWRKJTWLV3NM6RANCNFSM573AINGA]. You are receiving this because you are subscribed to this thread.[Image de pistage][https://github.com/notifications/beacon/AAAA4OWDLCHTKR3L5KUEPMTV3NM6RA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4UFKLURA.gif] Separated ifelse sections? — Reply to this email directly, view it on GitHub[[#325 (comment)](https://github.com/phhusson/device_phh_treble/pull/325#issuecomment-1229457612)], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4OR3CJX4GXJMUHAKMCDV3NSXXANCNFSM573AINGA]. You are receiving this because you commented.[Image de pistage][https://github.com/notifications/beacon/AAAA4OXSHNO3FQ5ZFIHMDBTV3NSXXA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOJFEARTA.gif]

Empty keylayout seems globally mounted for Xiaomi devices on [rw-system.sh](https://github.com/phhusson/device_phh_treble/blob/a4f61cfcbb4109e1a74fffa23a91704b6ccdac7a/rw-system.sh#L139)
Which working fine on a11 roms, but doesnt work on a12

phhusson commented:
Why?

28 août 2022 15:52:44 Zain Arbani ***@***.***>:

> No, mount binding key layout 28 août 2022 15:30:36 Zain Arbani /*@*/.///>:
> Please do it the same way it's done for other Xiaomi devices 28 août 2022 14:41:13 Zain Arbani //@//.//>: rosemary uses 96 for fp custom keycode which detected by InputManager as KEY_KPENTER. screenshots key combo will fail if more than 2 key events detected, exclude it. Signed-off-by: zainarbani //@//.*/ ---------------------------------------- /You can view, comment on, or merge this pull request online at:/   #325[https://github.com/phhusson/device_phh_treble/pull/325][#325[https://github.com/phhusson/device_phh_treble/pull/325]] /Commit Summary/
> 
* > ba81cfa[https://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6][ba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bba81cfa%5Bhttps://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6][https://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bba81cfa%5Bhttps://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6%5D]] Exclude FP input devices for rosemary
> /File Changes/ (2 files[#325[https://github.com/phhusson/device_phh_treble/pull/325]]/files])
> 
* > /A/ files/rosemary-excluded-input-devices.xml[#325[https://github.com/phhusson/device_phh_treble/pull/325]]/files#diff-083586316d8479c0b0d5c5996cb9ab961e60f5dad20d277f3b5c6d02a17b4dda] (31) * > /M/ rw-system.sh[#325[https://github.com/phhusson/device_phh_treble/pull/325]]/files#diff-29b2fa125a8bb33ed2b8f9ce4c2cc149006642b81294b5da700abbc058e21c95] (3)
> /Patch Links:/
> 
* > https://github.com/phhusson/device_phh_treble/pull/325.patch * > https://github.com/phhusson/device_phh_treble/pull/325.diff
> …[#]
> …[#] — Reply to this email directly, view it on GitHub[#325[https://github.com/phhusson/device_phh_treble/pull/325][#325[https://github.com/phhusson/device_phh_treble/pull/325]]], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4ORZ7NGQKNKKWRKJTWLV3NM6RANCNFSM573AINGA]. You are receiving this because you are subscribed to this thread.[Image de pistage][https://github.com/notifications/beacon/AAAA4OWDLCHTKR3L5KUEPMTV3NM6RA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4UFKLURA.gif] Separated ifelse sections? — Reply to this email directly, view it on GitHub[#325 (comment)[https://github.com/phhusson/device_phh_treble/pull/325#issuecomment-1229457612]], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4OR3CJX4GXJMUHAKMCDV3NSXXANCNFSM573AINGA]. You are receiving this because you commented.[Image de pistage][https://github.com/notifications/beacon/AAAA4OXSHNO3FQ5ZFIHMDBTV3NSXXA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOJFEARTA.gif]
> 
> Empty keylayout seems globally mounted for Xiaomi devices on rw-system.sh[https://github.com/phhusson/device_phh_treble/blob/a4f61cfcbb4109e1a74fffa23a91704b6ccdac7a/rw-system.sh#L139]
> Which working fine on a11 roms, but doesnt work on a12
> 
> —
> Reply to this email directly, view it on GitHub[https://github.com/phhusson/device_phh_treble/pull/325#issuecomment-1229462010], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4OSPP4VK4ZT7RP7ZK6DV3NVKZANCNFSM573AINGA].
> You are receiving this because you commented.[Image de pistage][https://github.com/notifications/beacon/AAAA4OSEUFXKBMBBF7LXM2DV3NVKZA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOJFEBT6Q.gif]
>

zainarbani commented:
> Why? 28 août 2022 15:52:44 Zain Arbani ***@***.***>:
> No, mount binding key layout 28 août 2022 15:30:36 Zain Arbani /*@*/.///>: Please do it the same way it's done for other Xiaomi devices 28 août 2022 14:41:13 Zain Arbani //@//.//>: rosemary uses 96 for fp custom keycode which detected by InputManager as KEY_KPENTER. screenshots key combo will fail if more than 2 key events detected, exclude it. Signed-off-by: zainarbani //@//.*/ ---------------------------------------- /You can view, comment on, or merge this pull request online at:/   #325[#325][#325[https://github.com/phhusson/device_phh_treble/pull/325]] /Commit Summary/
> * > [ba81cfa](https://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6)[[ba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bba81cfa%5Bhttps://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6][https://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6][ba81cfa[https://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6]](https://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bba81cfa%5Bhttps://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bhttps://github.com/phhusson/device_phh_treble/commit/ba81cfa6597167a4668f6a0940778a97972b0ed6%5D%5Bba81cfa%5Bhttps://github.com/phhusson/device_phh_treble/pull/325/commits/ba81cfa6597167a4668f6a0940778a97972b0ed6%5D)]] Exclude FP input devices for rosemary
> /File Changes/ (2 files[#325[#325]]/files])
> * > /A/ files/rosemary-excluded-input-devices.xml[#325[#325]]/files#diff-083586316d8479c0b0d5c5996cb9ab961e60f5dad20d277f3b5c6d02a17b4dda] (31) * > /M/ rw-system.sh[#325[#325]]/files#diff-29b2fa125a8bb33ed2b8f9ce4c2cc149006642b81294b5da700abbc058e21c95] (3)
> /Patch Links:/
> * > https://github.com/phhusson/device_phh_treble/pull/325.patch * > https://github.com/phhusson/device_phh_treble/pull/325.diff
> […](#)
> …[#] …[#] — Reply to this email directly, view it on GitHub[#325[#325][#325[https://github.com/phhusson/device_phh_treble/pull/325]]], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4ORZ7NGQKNKKWRKJTWLV3NM6RANCNFSM573AINGA]. You are receiving this because you are subscribed to this thread.[Image de pistage][https://github.com/notifications/beacon/AAAA4OWDLCHTKR3L5KUEPMTV3NM6RA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4UFKLURA.gif] Separated ifelse sections? — Reply to this email directly, view it on GitHub[#325 (comment)[[#325 (comment)](https://github.com/phhusson/device_phh_treble/pull/325#issuecomment-1229457612)]], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4OR3CJX4GXJMUHAKMCDV3NSXXANCNFSM573AINGA]. You are receiving this because you commented.[Image de pistage][https://github.com/notifications/beacon/AAAA4OXSHNO3FQ5ZFIHMDBTV3NSXXA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOJFEARTA.gif] Empty keylayout seems globally mounted for Xiaomi devices on rw-system.sh[https://github.com/phhusson/device_phh_treble/blob/a4f61cfcbb4109e1a74fffa23a91704b6ccdac7a/rw-system.sh#L139] Which working fine on a11 roms, but doesnt work on a12 — Reply to this email directly, view it on GitHub[[#325 (comment)](https://github.com/phhusson/device_phh_treble/pull/325#issuecomment-1229462010)], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AAAA4OSPP4VK4ZT7RP7ZK6DV3NVKZANCNFSM573AINGA]. You are receiving this because you commented.[Image de pistage][https://github.com/notifications/beacon/AAAA4OSEUFXKBMBBF7LXM2DV3NVKZA5CNFSM573AINGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOJFEBT6Q.gif]

Im not sure which cause that on a12, wether device specific or any devices that share same linux common [keycodes](https://github.com/aosp-mirror/platform_bionic/blob/master/libc/kernel/uapi/linux/input-event-codes.h).
This issue also exist on non GSI build, and this is my current workaround for that.
I can confirm that fingerprint enrollments/auth still works fine, FP gesture or related feature was not tested.

iovxw commented on rw-system.sh:
```diff
@@ -585,6 +585,9 @@ if getprop ro.vendor.build.fingerprint | grep -iq -e Redmi/rosemary \
     setprop debug.sf.enable_hwc_vds 0
+
+    # Exclude FP input devices
+    mount -o bind /system/phh/rosemary-excluded-input-devices.xml /system/etc/excluded-input-devices.xml
```
I don't think this will work, have you updated the `PRODUCT_COPY_FILES`?

zainarbani replied:
Updated on https://github.com/phhusson/device_phh_treble/pull/325/commits/091321c413955c9d2015ec39871aae4da506548b

iovxw replied:
Have you tested it? `/system/etc/excluded-input-devices.xml` doesn't exist the mount bind will fail

iovxw commented:
gauguin have the same issue, side-mounted fingerprint sensor, to take a screenshot I have to use my nail to press the power button

The mount binding key layout does work, no event is sent to applications, but not the key combo

zainarbani committed:
Exclude FP input devices for rosemary

 rosemary uses 96 for fp custom keycode which detected by InputManager as KEY_KPENTER.
 screenshots key combo will fail if more than 2 key events detected, exclude it.

Signed-off-by: zainarbani <zaintsyariev@gmail.com>
